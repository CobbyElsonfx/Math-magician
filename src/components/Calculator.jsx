import React, { useState } from 'react';
// import Screen from './Screen';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = async (buttonName) => {
    // setCalculatorData()
    const newData = await calculate(calculatorData, buttonName);
    setCalculatorData(newData);
  };

  return (
    <>
      <div className="calculatorWrapper">
        <div className="displayScreen">
          <div>
            <span className="calculatorValueOnScreen">{calculatorData.next || calculatorData.total || 0}</span>
          </div>
        </div>
        <div className="buttonsContainer">
          <button type="button" onClick={() => handleClick('AC')}>AC</button>
          <button type="button" onClick={() => handleClick('+')}>+/-</button>
          <button type="button" onClick={() => handleClick('%')}>%</button>
          <button type="button" onClick={() => handleClick('รท')}>&divide;</button>
        </div>
        <div className="buttonsContainer">
          <button type="button" onClick={() => handleClick('7')}>7</button>
          <button type="button" onClick={() => handleClick('8')}>8</button>
          <button type="button" onClick={() => handleClick('9')}>9</button>
          <button type="button" onClick={() => handleClick('x')}>&times;</button>
        </div>
        <div className="buttonsContainer">
          <button type="button" onClick={() => handleClick('4')}>4</button>
          <button type="button" onClick={() => handleClick('5')}>5</button>
          <button type="button" onClick={() => handleClick('6')}>6</button>
          <button type="button" onClick={() => handleClick('-')}>-</button>
        </div>
        <div className="buttonsContainer">
          <button type="button" onClick={() => handleClick('1')}>1</button>
          <button type="button" onClick={() => handleClick('2')}>2</button>
          <button type="button" onClick={() => handleClick('3')}>3</button>
          <button type="button" onClick={() => handleClick('+')}>+</button>
        </div>
        <div className="buttonsContainer">
          <button type="button" id="zeroButton" onClick={() => handleClick('AC')}>0</button>
          <button type="button" onClick={() => handleClick('.')}>.</button>
          <button type="button" onClick={() => handleClick('=')}>=</button>
        </div>
      </div>
    </>
  );
};

export default Calculator;
