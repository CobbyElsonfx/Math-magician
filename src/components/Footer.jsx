const Footer = () => (
  <div>
    <footer className="footer">
      <p>
        Created by
        <a href="./">CobbyElsonfx</a>
      </p>
    </footer>
  </div>
);

export default Footer;
