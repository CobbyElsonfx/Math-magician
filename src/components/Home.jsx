const Home = () => (
  <div className="homePageWrapper">
    <div className="sideBars">
      <div className="sideBar" />
      <div className="sideBarFull" />
    </div>
    <div className="homeImg">
      <h1>
        Welcome to&nbsp;
        <span className="MathMagicianWelc">Maths Magician</span>
      </h1>
      <p className="pageIntro">
        Get ready to explore the world of mathematics. Whether you are looking
        for quick calculations or inspiring math quotes, we have got you covered.
        Lets start your mathematical journey!
      </p>
    </div>
  </div>
);

export default Home;
