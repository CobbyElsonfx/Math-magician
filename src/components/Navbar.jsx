const Navbar = () => (
  <nav className="navbar">
    <div className="logo">
      <a href="./">
        M
        {' '}
        <span className="symb">=+^</span>
      </a>
    </div>
    <div className="links">
      <ul>
        <li><a href="/">Home</a></li>
        <li><a href="/calculator">Calculator</a></li>
        <li><a href="/quote">Quote</a></li>
      </ul>
    </div>
  </nav>
);

export default Navbar;
