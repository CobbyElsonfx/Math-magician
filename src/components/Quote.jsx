import React, { useState, useEffect } from 'react';

const Quote = () => {
  const [quotes, setQuotes] = useState('');
  const [error, setError] = useState('');
  const [isLoading, setIsLoading] = useState(true);

  const fetchData = async () => {
    const quotesFetch = await fetch('https://api.api-ninjas.com/v1/quotes?category=learning', {
      method: 'GET',
      headers: {
        'X-Api-Key': 'GZDTbliwsEhcjqbDy6PgYA==ZPoFVCTEyv9xsIsq',
        'Content-Type': 'application/json',
      },
    });

    const quotesJson = await quotesFetch.json();
    if (!quotesJson.ok) {
      setError('Something went wrong');
    }
    setQuotes(quotesJson);
    setIsLoading(false);
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <div>
      {isLoading ? <p>Loading...</p>
        : (
          <div>
            <p>{quotes[0].quote}</p>
            <small>{quotes[0].author}</small>
          </div>
        )}
      {error && <p>{error}</p>}
    </div>
  );
};

export default Quote;
