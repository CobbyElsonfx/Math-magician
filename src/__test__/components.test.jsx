import { render } from '@testing-library/react';
import Home from '../components/Home';
import Navbar from '../components/Navbar';
import Calculator from '../components/Calculator';
import Footer from '../components/Footer';
import Quote from '../components/Quote';

describe('Testing Calculator Components', () => {
  test('should render home component', () => {
    const home = render(<Home />);
    expect(home).toMatchSnapshot();
  });

  test('should render Navbar', () => {
    const navbar = render(<Navbar />);
    expect(navbar).toMatchSnapshot();
  });

  test('should render Calculator', () => {
    const calculator = render(<Calculator />);
    expect(calculator).toMatchSnapshot();
  });

  test('should render Footer', () => {
    const footer = render(<Footer />);
    expect(footer).toMatchSnapshot();
  });

  test('should render Quote', () => {
    const quote = render(<Quote />);
    expect(quote).toMatchSnapshot();
  });
});
