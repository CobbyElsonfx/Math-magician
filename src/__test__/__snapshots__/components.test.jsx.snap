// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calculator Components Testing Home Component testing Home Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="homePageWrapper"
      >
        <div
          class="sideBars"
        >
          <div
            class="sideBar"
          />
          <div
            class="sideBarFull"
          />
        </div>
        <div
          class="homeImg"
        >
          <h1>
            Welcome to 
            <span
              class="MathMagicianWelc"
            >
              Maths Magician
            </span>
          </h1>
          <p
            class="pageIntro"
          >
            Get ready to explore the world of mathematics. Whether you are looking for quick calculations or inspiring math quotes, we have got you covered. Lets start your mathematical journey!
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="homePageWrapper"
    >
      <div
        class="sideBars"
      >
        <div
          class="sideBar"
        />
        <div
          class="sideBarFull"
        />
      </div>
      <div
        class="homeImg"
      >
        <h1>
          Welcome to 
          <span
            class="MathMagicianWelc"
          >
            Maths Magician
          </span>
        </h1>
        <p
          class="pageIntro"
        >
          Get ready to explore the world of mathematics. Whether you are looking for quick calculations or inspiring math quotes, we have got you covered. Lets start your mathematical journey!
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Calculator Components should render Calculator 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculatorWrapperFall"
      >
        <div>
          <h2>
            Lets do some Maths
          </h2>
        </div>
        <div
          class="calculatorWrapper"
        >
          <div
            class="displayScreen"
          >
            <div>
              <span
                class=" calculatorValueOnScreen"
              >
                0
              </span>
            </div>
          </div>
          <div
            class="buttonsContainer"
          >
            <button
              type="button"
            >
              AC
            </button>
            <button
              type="button"
            >
              +/-
            </button>
            <button
              type="button"
            >
              %
            </button>
            <button
              type="button"
            >
              ÷
            </button>
          </div>
          <div
            class="buttonsContainer"
          >
            <button
              type="button"
            >
              7
            </button>
            <button
              type="button"
            >
              8
            </button>
            <button
              type="button"
            >
              9
            </button>
            <button
              type="button"
            >
              ×
            </button>
          </div>
          <div
            class="buttonsContainer"
          >
            <button
              type="button"
            >
              4
            </button>
            <button
              type="button"
            >
              5
            </button>
            <button
              type="button"
            >
              6
            </button>
            <button
              type="button"
            >
              -
            </button>
          </div>
          <div
            class="buttonsContainer"
          >
            <button
              type="button"
            >
              1
            </button>
            <button
              type="button"
            >
              2
            </button>
            <button
              type="button"
            >
              3
            </button>
            <button
              type="button"
            >
              +
            </button>
          </div>
          <div
            class="buttonsContainer"
          >
            <button
              id="zeroButton"
              type="button"
            >
              0
            </button>
            <button
              type="button"
            >
              .
            </button>
            <button
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculatorWrapperFall"
    >
      <div>
        <h2>
          Lets do some Maths
        </h2>
      </div>
      <div
        class="calculatorWrapper"
      >
        <div
          class="displayScreen"
        >
          <div>
            <span
              class=" calculatorValueOnScreen"
            >
              0
            </span>
          </div>
        </div>
        <div
          class="buttonsContainer"
        >
          <button
            type="button"
          >
            AC
          </button>
          <button
            type="button"
          >
            +/-
          </button>
          <button
            type="button"
          >
            %
          </button>
          <button
            type="button"
          >
            ÷
          </button>
        </div>
        <div
          class="buttonsContainer"
        >
          <button
            type="button"
          >
            7
          </button>
          <button
            type="button"
          >
            8
          </button>
          <button
            type="button"
          >
            9
          </button>
          <button
            type="button"
          >
            ×
          </button>
        </div>
        <div
          class="buttonsContainer"
        >
          <button
            type="button"
          >
            4
          </button>
          <button
            type="button"
          >
            5
          </button>
          <button
            type="button"
          >
            6
          </button>
          <button
            type="button"
          >
            -
          </button>
        </div>
        <div
          class="buttonsContainer"
        >
          <button
            type="button"
          >
            1
          </button>
          <button
            type="button"
          >
            2
          </button>
          <button
            type="button"
          >
            3
          </button>
          <button
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="buttonsContainer"
        >
          <button
            id="zeroButton"
            type="button"
          >
            0
          </button>
          <button
            type="button"
          >
            .
          </button>
          <button
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Calculator Components should render Footer 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <footer
          class="footer"
        >
          <p>
            Created by
            <a
              href="./"
            >
              CobbyElsonfx
            </a>
          </p>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <footer
        class="footer"
      >
        <p>
          Created by
          <a
            href="./"
          >
            CobbyElsonfx
          </a>
        </p>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Calculator Components should render Navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar"
      >
        <div
          class="logo"
        >
          <a
            href="./"
          >
            M
             
            <span
              class="symb"
            >
              =+^
            </span>
          </a>
        </div>
        <div
          class="links"
        >
          <ul>
            <li>
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/calculator"
              >
                Calculator
              </a>
            </li>
            <li>
              <a
                href="/quote"
              >
                Quote
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar"
    >
      <div
        class="logo"
      >
        <a
          href="./"
        >
          M
           
          <span
            class="symb"
          >
            =+^
          </span>
        </a>
      </div>
      <div
        class="links"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <li>
            <a
              href="/quote"
            >
              Quote
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Calculator Components should render Quote 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="quoteWrapper"
      >
        <hr />
        <p>
          Loading...
        </p>
        <hr />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="quoteWrapper"
    >
      <hr />
      <p>
        Loading...
      </p>
      <hr />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Calculator Components should render home component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="homePageWrapper"
      >
        <div
          class="sideBars"
        >
          <div
            class="sideBar"
          />
          <div
            class="sideBarFull"
          />
        </div>
        <div
          class="homeImg"
        >
          <h1>
            Welcome to 
            <span
              class="MathMagicianWelc"
            >
              Maths Magician
            </span>
          </h1>
          <p
            class="pageIntro"
          >
            Get ready to explore the world of mathematics. Whether you are looking for quick calculations or inspiring math quotes, we have got you covered. Lets start your mathematical journey!
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="homePageWrapper"
    >
      <div
        class="sideBars"
      >
        <div
          class="sideBar"
        />
        <div
          class="sideBarFull"
        />
      </div>
      <div
        class="homeImg"
      >
        <h1>
          Welcome to 
          <span
            class="MathMagicianWelc"
          >
            Maths Magician
          </span>
        </h1>
        <p
          class="pageIntro"
        >
          Get ready to explore the world of mathematics. Whether you are looking for quick calculations or inspiring math quotes, we have got you covered. Lets start your mathematical journey!
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
